python/tank/bootstrap/resolver.py

```diff
@@ -166,11 +129,43 @@ def resolve_configuration(
         # create an object to represent our configuration install
         return Configuration(
             config_root,
-            self._sg_connection,
+            sg_connection,
             cfg_descriptor,
-            project_id,
+            self._project_id,
             None,  # pipeline config id
             self._bundle_cache_fallback_paths
         )
 
+    def resolve_shotgun_configuration(
+        self,
+        pipeline_config_name,
+        fallback_config_descriptor,
+        sg_connection
+    ):
+        """
+        Return a configuration object by requesting a pipeline configuration
+        in Shotgun. If no suitable configuration is found, return a configuration
+        for the given fallback config.
+
+        :param pipeline_config_name: Name of configuration branch (e.g Primary)
+        :param fallback_config_descriptor: descriptor dict or string for fallback config.
+        :param sg_connection: Shotgun API instance
+        :return: :class:`Configuration` instance
+        """
+        log.debug(
+            "%s resolving configuration from Shotgun Pipeline Configuration %s" % (self, pipeline_config_name)
+        )
+
+        log.warning(
+            "Shotgun pipeline configuration resolve has not been implemented yet. "
+            "Falling back on the base configuration resolver."
+        )
```

Comment:
Ok.  I think we should update the docstring to reflect that then.  So that if people are browsing via Sphinx they know what the state of this implementation is.